modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/leases/LeaseTracker.java

```diff
@@ -162,23 +176,28 @@ public CompletableFuture<Void> onUpdate(WatchEvent event) {
             for (EntryEvent entry : event.entryEvents()) {
                 Entry msEntry = entry.newEntry();
 
-                String key = new ByteArray(msEntry.key()).toString();
-
-                key = key.replace(PLACEMENTDRIVER_PREFIX, "");
+                LeaseBatch leaseBatch = LeaseBatch.fromBytes(ByteBuffer.wrap(msEntry.value()).order(ByteOrder.LITTLE_ENDIAN));
 
-                TablePartitionId grpId = TablePartitionId.fromString(key);
+                Set<ReplicationGroupId> actualGroups = new HashSet<>();
 
-                if (msEntry.empty()) {
-                    leases.remove(grpId);
-                    tryRemoveTracker(grpId);
-                } else {
-                    Lease lease = fromBytes(msEntry.value());
+                for (Lease lease : leaseBatch.leases()) {
+                    ReplicationGroupId grpId = lease.replicationGroupId();
+                    actualGroups.add(grpId);
 
                     leases.put(grpId, lease);
 
                     primaryReplicaWaiters.computeIfAbsent(grpId, groupId -> new PendingIndependentComparableValuesTracker<>(MIN_VALUE))
                             .update(lease.getExpirationTime(), lease);
```

Comment:
Because now we have the same expiration timestamp for all leases in batch, we will be able to got rid of that waiters (replace them to one).

Reply:
This should be done under https://issues.apache.org/jira/browse/IGNITE-19819